File: android-native/app/src/main/java/com/ironcore/fit/data/model/Tournament.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: /tournaments/{tournamentId}
 * Also mirrored at /global/tournament for active singleton.
 */
data class Tournament(
    val id: String = "",
    val title: String = "",
    val description: String = "",
    val startDate: String = "",
    val endDate: String = "",
    val status: String = "upcoming",
    val rules: String = "",
    val rewards: List<TournamentReward> = emptyList(),
    val participantCount: Int = 0,
    @ServerTimestamp val createdAt: Timestamp? = null
)

data class TournamentReward(
    val rank: Int = 0,
    val reward: String = ""
)

/**
 * Firestore path: /tournaments/{id}/participants/{userId}
 */
data class TournamentParticipant(
    val userId: String = "",
    val username: String = "",
    val avatarUrl: String = "",
    val score: Long = 0,
    val rank: Int = 0,
    val joinedAt: Timestamp? = null
)

File: ios-native/IronCore/Models/Tournament.swift
import Foundation
import FirebaseFirestore

/// Tournament at `tournaments/{tournamentId}`
struct Tournament: Codable, Identifiable {
    @DocumentID var id: String?
    var title: String
    var description: String
    var startDate: String       // ISO8601
    var endDate: String         // ISO8601
    var status: String          // "upcoming", "active", "completed"
    var rules: String
    var rewards: [TournamentReward]
    var participantCount: Int
    @ServerTimestamp var createdAt: Timestamp?
}

struct TournamentReward: Codable {
    var rank: Int
    var reward: String
}

/// Participant at `tournaments/{tournamentId}/participants/{userId}`
struct TournamentParticipant: Codable, Identifiable {
    @DocumentID var id: String?
    var userId: String
    var username: String
    var avatarUrl: String
    var score: Int
    var rank: Int
    @ServerTimestamp var joinedAt: Timestamp?
}
